src/services/fetchData.ts

```diff
@@ -22,20 +22,29 @@ const fetchProducts = async (param?: string): Promise<Product[]> => {
   }
 };
 
-const fetchSearchProducts = async (params: SearchProductsProps): Promise<Product[]> => {
+const fetchProductsParams = async (params: FilterProductsProps): Promise<Product[]> => {
   try {
     const matchNameProducts = await fetchProducts(`name_like=${params.nameSearch}`);
     const matchDescProducts = await fetchProducts(`description_like=${params.descSearch}`);
-    const allProducts = [...matchNameProducts, ...matchDescProducts];
 
-    const uniqueProducts = allProducts.filter(
+    const allSearchProducts = [...matchNameProducts, ...matchDescProducts];
+    const uniqueSearchProducts = allSearchProducts.filter(
       (product, index, self) => index === self.findIndex((p) => p.objectID === product.objectID),
     );
 
-    return uniqueProducts;
+    const matchSortProducts = await fetchProducts(`_sort=${params.sortBy}&_order=${params.sortOrder || 'asc'}`);
```

Comment:
E xử lý params 1 lần rồi gọi 1 lần api thôi nha, không gọi 3 lần như hiện tại (vì json server không hỗ trợ cho or nên mình dùng & 2 điều kiện. Thực tế thì phần này ở BE sẽ lo cho mình)
Bổ sung thêm filter limit và page để phân trang luôn
https://www.npmjs.com/package/json-server/v/0.17.4#:~:text=GET%20/posts%3F_page%3D7%0AGET%20/posts%3F_page%3D7%26_limit%3D20